pkg/dependency/parser/python/binary/exe.go

```diff
@@ -0,0 +1,82 @@
+// Ported from https://github.com/golang/go/blob/e9c96835971044aa4ace37c7787de231bbde05d9/src/cmd/go/internal/version/exe.go
+
+package binary
+
+import (
+	"bytes"
+	"debug/elf"
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+// An exe is a generic interface to an OS executable (ELF, Mach-O, PE, XCOFF).
+type exe interface {
+	// ReadData reads and returns up to size byte starting at virtual address addr.
+	ReadData(addr, size uint64) ([]byte, error)
+
+	// DataStart returns the writable data segment start address.
+	DataStart() (uint64, uint64)
+}
+
+// openExe opens file and returns it as an exe.
+func openExe(r io.Reader) (exe, error) {
+	b, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+
+	br := bytes.NewReader(b)
+
+	data := make([]byte, 16)
+	if _, err := io.ReadFull(br, data); err != nil {
+		return nil, err
+	}
+	_, err = br.Seek(0, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	if bytes.HasPrefix(data, []byte("\x7FELF")) {
```

Comment:
Haven't thought about that yet. Will have to check with those.

Reply:
I tried testing with Windows PEs, the interface can be implemented for openExe, but the regex based detection methods don't work on the PE files I tried (I did manual testing to confirm). I honestly don't have any experience with macos binaries 🫤 So not sure if I can implement the interface for those. I also tried to obtain files to test with, but failed, since now I don't have access to a macos system. Any idea if regex based detection would work on those?

If not, I'd rather not bother with implementing this for those binaries, even though you could implement the interface and `executableAnalyzer` would still run on those, the binary version detection won't give package in the `AnalysisResult`.